lib/modules/versioning/azure-rest-api/index.ts

```diff
@@ -0,0 +1,49 @@
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'azure-rest-api';
+export const displayName = 'azure-rest-api';
+
+export const urls = [
+  'https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-versioning',
+];
+
+export const supportsRanges = false;
+
+const AZURE_REST_API_VERSION_REGEX = regEx(
+  /^(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})(?<prerelease>-[a-z]+)?$/
```

Comment:
Can this be stricter? prerelease is always going to be `-preview` when defined?

Reply:
Actually, I just saw your comment below about the additional suffixes :sweat_smile: